src/ui/public/embeddable/embeddable_factory.ts

```diff
@@ -43,7 +33,8 @@ export class EmbeddableFactory {
    * state whenever something changes that the dashboard should know about.
    * @return {Promise.<Embeddable>}
    */
-  create(/* containerMetadata, onEmbeddableStateChanged*/) {
-    throw new Error('Must implement create.');
-  }
+  public abstract create(
+    containerMetadata: object,
```

Comment:
I don't think that's quite right - it's not a mapping.  So right now we pass in PanelState because saved search and visualize embeddables use `containerMetadata.id` to create the saved object - that's actually all they care about so we could just pass in id.  But this leaks knowledge of embeddable internals - that there is an instance of something to be added.

We haven't yet hooked up the part of generalizing and supporting embeddables in the "add panel" part of dashboard but I didn't want to assume every embeddable is a 1:1 mapping to a saved object with an id.

But maybe I'm thinking too far into the future.  I suppose for now I can just assume that every create call requires an id of some sort.